[write-behind]:Remove unused function

pbhatia22 opened the pull request:
Remove unused function
1.__wb_request_ref used directly instead of wb_request_ref

2.wb_open & wb_create-> dead code

Change-Id: I057eb6bd818802a00f21ffe5418b1f181916eff4
Updates:#1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

pbhatia22 committed:
[write-behind]:Remove unused function

1.__wb_request_ref used directly instead of wb_request_ref

2.wb_open & wb_create-> dead code

Change-Id: I057eb6bd818802a00f21ffe5418b1f181916eff4
Updates:#1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/performance/write-behind/src/write-behind.c b/xlators/performance/write-behind/src/write-behind.c
index 5517f50ee..83720130c 100644
--- a/xlators/performance/write-behind/src/write-behind.c
+++ b/xlators/performance/write-behind/src/write-behind.c
@@ -55,10 +55,10 @@ typedef struct wb_inode {
                                which arrive later (which overlap, etc.)
                                are issued only after their dependencies
                                in this list are "fulfilled".
-
+ 
                                Server acks for entries in this list
                                shrinks the window.
-
+ 
                                The sum total of all req->write_size
                                of entries in this list must be kept less
                                than the permitted window size.
@@ -92,14 +92,14 @@ typedef struct wb_inode {
                                     the current 'state' of liability. Every
                                     new addition to the liability list bumps
                                     the generation number.
-
+               
                                     a newly arrived request is only required
                                     to perform causal checks against the entries
                                     in the liability list which were present
                                     at the time of its addition. the generation
                                     number at the time of its addition is stored
                                     in the request and used during checks.
-
+               
                                     the liability list can grow while the request
                                     waits in the todo list waiting for its
                                     dependent operations to complete. however
@@ -486,7 +486,6 @@ out:
     return req;
 }
 
-
 gf_boolean_t
 wb_enqueue_common(wb_inode_t *wb_inode, call_stub_t *stub, int tempted)
 {
```

mykaul commented:
nit: I think you can remove the vadlidation from __wb_request_ref() 
```
    GF_VALIDATE_OR_GOTO("write-behind", req, out);
```

As all callers already either check or use req.

pbhatia22 committed:
[write-behind]:Remove unused function

1.__wb_request_ref used directly instead of wb_request_ref

2.wb_open & wb_create-> dead code

Change-Id: I057eb6bd818802a00f21ffe5418b1f181916eff4
Updates:gluster#1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/performance/write-behind/src/write-behind.c b/xlators/performance/write-behind/src/write-behind.c
index d7d027e82..efb14bf72 100644
--- a/xlators/performance/write-behind/src/write-behind.c
+++ b/xlators/performance/write-behind/src/write-behind.c
@@ -55,10 +55,10 @@ typedef struct wb_inode {
                                which arrive later (which overlap, etc.)
                                are issued only after their dependencies
                                in this list are "fulfilled".
-
+ 
                                Server acks for entries in this list
                                shrinks the window.
-
+ 
                                The sum total of all req->write_size
                                of entries in this list must be kept less
                                than the permitted window size.
@@ -92,14 +92,14 @@ typedef struct wb_inode {
                                     the current 'state' of liability. Every
                                     new addition to the liability list bumps
                                     the generation number.
-
+               
                                     a newly arrived request is only required
                                     to perform causal checks against the entries
                                     in the liability list which were present
                                     at the time of its addition. the generation
                                     number at the time of its addition is stored
                                     in the request and used during checks.
-
+               
                                     the liability list can grow while the request
                                     waits in the todo list waiting for its
                                     dependent operations to complete. however
@@ -470,7 +470,6 @@ out:
 static wb_request_t *
 __wb_request_ref(wb_request_t *req)
 {
-
     if (req->refcount < 0) {
         gf_msg("wb-request", GF_LOG_WARNING, 0,
                WRITE_BEHIND_MSG_RES_UNAVAILABLE, "refcount(%d) is < 0",
```

xhernandez commented on xlators/performance/write-behind/src/write-behind.c:
```diff
@@ -470,7 +470,6 @@ wb_request_unref(wb_request_t *req)
 __wb_request_ref(wb_request_t *req)
 {
-    GF_VALIDATE_OR_GOTO("write-behind", req, out);
 
```
nit: please, remove the empty line.

pbhatia22 committed:
[write-behind]:Remove unused function

1.__wb_request_ref used directly instead of wb_request_ref

2.wb_open & wb_create-> dead code

Change-Id: I057eb6bd818802a00f21ffe5418b1f181916eff4
Updates:gluster#1000
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/performance/write-behind/src/write-behind.c b/xlators/performance/write-behind/src/write-behind.c
index e282d5b9b..efb14bf72 100644
--- a/xlators/performance/write-behind/src/write-behind.c
+++ b/xlators/performance/write-behind/src/write-behind.c
@@ -55,10 +55,10 @@ typedef struct wb_inode {
                                which arrive later (which overlap, etc.)
                                are issued only after their dependencies
                                in this list are "fulfilled".
-
+ 
                                Server acks for entries in this list
                                shrinks the window.
-
+ 
                                The sum total of all req->write_size
                                of entries in this list must be kept less
                                than the permitted window size.
@@ -92,14 +92,14 @@ typedef struct wb_inode {
                                     the current 'state' of liability. Every
                                     new addition to the liability list bumps
                                     the generation number.
-
+               
                                     a newly arrived request is only required
                                     to perform causal checks against the entries
                                     in the liability list which were present
                                     at the time of its addition. the generation
                                     number at the time of its addition is stored
                                     in the request and used during checks.
-
+               
                                     the liability list can grow while the request
                                     waits in the todo list waiting for its
                                     dependent operations to complete. however
```

pbhatia22 commented:
/run regression
